File: task/src/main/java/com/goexp/galgame/data/model/Game.java
package com.goexp.galgame.data.model;

import com.goexp.galgame.common.model.CommonGame;
import com.goexp.galgame.common.model.GameState;

import java.util.Objects;

public class Game extends CommonGame {


    public GameState state;

    public int brandId;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Game)) return false;
        Game game = (Game) o;
        return id == game.id &&
//                Objects.equals(name, game.name) &&
//                Objects.equals(publishDate, game.publishDate) &&
//                Objects.equals(imgUrl, game.imgUrl) &&
//                Objects.equals(website, game.website) &&
                Objects.equals(brandId, game.brandId) &&
                Objects.equals(writer, game.writer) &&
                Objects.equals(painter, game.painter) &&
                Objects.equals(type, game.type) &&
                Objects.equals(tag, game.tag) &&
                Objects.equals(story, game.story);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id,
//                name,
//                publishDate,
//                imgUrl,
                brandId,
                writer,
                painter,
                type,
                tag,
                story);
    }


}

File: task/src/main/scala/com/goexp/galgame/data/model/Game.scala
package com.goexp.galgame.data.model

import java.util.{Objects, StringJoiner}

import com.goexp.common.util.string.ConsoleColors.RED
import com.goexp.galgame.common.model.game.{CommonGame, GameState}

class Game extends CommonGame {
  var state: GameState = _
  var brandId = 0
  var group: String = _

  override def equals(o: Any): Boolean = {

    o match {
      case game: Game =>
        id == game.id &&
          (isAdult == game.isAdult) &&
          (brandId == game.brandId) &&
          (writer == game.writer) &&
          (painter == game.painter) &&
          (`type` == game.`type`) &&
          (tag == game.tag) &&
          (story == game.story)
      case _ => false
    }
  }

  def simpleView: String = "Game{" + "id=" + RED.s(id.toString) + ", name='" + RED.s(name) + '\'' + ", publishDate=" + publishDate + ", imgUrl='" + smallImg + '}'


  override def toString: String = {
    new StringJoiner(", ", classOf[Game].getSimpleName + "[", "]")
      .add("id=" + id)
      .add("isAdult='" + isAdult + "'")
      .add("name='" + RED.s(name) + "'")
      .add("publishDate=" + publishDate)
      .add("smallImg='" + smallImg + "'")
      //      .add("writer=" + writer)
      //      .add("painter=" + painter)
      //      .add("type=" + `type`)
      //      .add("tag=" + tag)
      .add("gameImgs=" + gameImgs)
      .add("gameCharacters=" + gameCharacters)
      //                .add("story='" + story + "'")
      //                .add("intro='" + intro + "'")
      .toString
  }

  override def hashCode: Int =
    Objects.hash(
      id.asInstanceOf,
      isAdult.asInstanceOf,
      brandId.asInstanceOf,
      writer,
      painter,
      `type`,
      tag,
      story)

}